File: backend/src/main/java/lan/dk/podcastserver/manager/worker/jeuxvideocom/JeuxVideoComFinder.java
package lan.dk.podcastserver.manager.worker.jeuxvideocom;

import com.github.davinkevin.podcastserver.service.HtmlService;
import lan.dk.podcastserver.entity.Podcast;
import lan.dk.podcastserver.manager.worker.Finder;
import lombok.RequiredArgsConstructor;
import org.apache.commons.lang3.StringUtils;
import org.jsoup.nodes.Document;
import org.springframework.stereotype.Service;

import java.util.function.Function;

/**
 * Created by kevin on 22/03/2016 for Podcast Server
 */
@Service
@RequiredArgsConstructor
public class JeuxVideoComFinder implements Finder {

    private final HtmlService htmlService;

    @Override
    public Podcast find(String url) {
        return htmlService
                .get(url)
                .map(htmlToPodcast(url))
                .getOrElse(Podcast.DEFAULT_PODCAST);
    }

    private Function<Document, Podcast> htmlToPodcast(String url) {
        return p -> Podcast.builder()
                    .title(p.select("meta[property=og:title]").attr("content"))
                    .description(p.select("meta[name=description]").attr("content"))
                    .url(url)
                    .type("JeuxVideoCom")
                .build();
    }

    @Override
    public Integer compatibility(String url) {
        return StringUtils.contains(url, "jeuxvideo.com") ? 1 : Integer.MAX_VALUE;
    }
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/manager/worker/jeuxvideocom/JeuxVideoComFinder.kt
package com.github.davinkevin.podcastserver.manager.worker.jeuxvideocom

import arrow.core.getOrElse
import com.github.davinkevin.podcastserver.service.HtmlService
import com.github.davinkevin.podcastserver.utils.k
import lan.dk.podcastserver.entity.Podcast
import lan.dk.podcastserver.manager.worker.Finder
import org.jsoup.nodes.Document
import org.springframework.stereotype.Service

/**
 * Created by kevin on 22/03/2016 for Podcast Server
 */
@Service
class JeuxVideoComFinder(val htmlService: HtmlService) : Finder {

    override fun find(url: String): Podcast {
        return htmlService.get(url).k()
                .map { htmlToPodcast(url, it) }
                .getOrElse { Podcast.DEFAULT_PODCAST }
    }

    private fun htmlToPodcast(u: String, d: Document): Podcast {
        return Podcast().apply {
            title = d.select("meta[property=og:title]").attr("content")
            description = d.select("meta[name=description]").attr("content")
            url = u
            type = "JeuxVideoCom"
        }
    }

    override fun compatibility(url: String?) = JeuxVideoComUpdater.isFromJeuxVideoCom(url)
}
